### PR TITLE
Profile basic info section

### PR DESCRIPTION
**Pull request by ashlilili**
1. Profile basic info section; displays vacation banner, user type, affiliations, company, education, etc.
2. moved three dots menu for block/report user to a separate file for code reuse
3. extended profileChip class to have an option to use primary color

Note: 
The vacation banner color may seem odd with profile page header, as figma uses primary95, which is not in the color scheme, so I'm using the closest color primaryContainer instead. I will follow up with Katy on this.

---

https://github.com/micromentor-team/mm-flutter-app/assets/100324646/5a6c30f5-389e-4b5a-973b-579d2227d13f

All fields
----

Screenshot 2023-08-02 at 5 04 51 PM
Only mandatory fields

### CODE REVIEW

File: lib/widgets/screens/profile/profile_basic_info.dart [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+import 'package:mm_flutter_app/constants/app_constants.dart';
+import 'package:mm_flutter_app/utilities/scaffold_utils/report_or_block_menu_button.dart';
+import 'package:mm_flutter_app/widgets/atoms/profile_chip.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+const double _inlineIconSize = 16.0;
+
+Future<void> _launchUrl(Uri url) async {
+  if (!await launchUrl(url)) {
+    throw Exception('Could not launch $url');
```
Comment (rupalg) [inline lib/widgets/screens/profile/profile_basic_info.dart:12]: Wow, great to see error handling!

File: lib/widgets/screens/profile/profile_basic_info.dart [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+            child: SizedBox(
+              width: _inlineIconSize,
+              child: Image(
+                image: AssetImage(Assets.linkedInIcon),
+              ),
+            ),
+          ),
+          onPressed: () {
+            debugPrint(linkedinUrl);
+            _launchUrl(Uri.parse(linkedinUrl));
```
Comment (rupalg) [inline lib/widgets/screens/profile/profile_basic_info.dart:87]: Curious: how does this work when you have the app? Does it open the linkedIn url on the browser or within the app?

**Review by rupalg — COMMENTED** [review]
Great work, Ashley! I really liked how the profile header is so customizable. Left a couple changes, mostly nits.

File: lib/widgets/screens/profile/profile_basic_info.dart [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+      color: theme.colorScheme.primaryContainer,
+      height: 48.0,
+      child: Row(mainAxisAlignment: MainAxisAlignment.center, children: [
+        Wrap(
+          crossAxisAlignment: WrapCrossAlignment.center,
+          children: [
+            const Icon(
+              Icons.beach_access_outlined,
+              size: _inlineIconSize,
+            ),
```
Comment (m-rosario) [inline lib/widgets/screens/profile/profile_basic_info.dart:155]: This Icon should be the same color as the text (onPrimaryContainer).

File: lib/widgets/screens/profile/profile_basic_info.dart [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:mm_flutter_app/constants/app_constants.dart';
+import 'package:mm_flutter_app/utilities/scaffold_utils/report_or_block_menu_button.dart';
+import 'package:mm_flutter_app/widgets/atoms/profile_chip.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+const double _inlineIconSize = 16.0;
+
+Future<void> _launchUrl(Uri url) async {
+  if (!await launchUrl(url)) {
```
Comment (m-rosario) [inline lib/widgets/screens/profile/profile_basic_info.dart:11]: Definitively out of scope for this PR, but launching a URL like this basically let's LinkedIn take over our app. The only way the user can go back to our main app is by pressing back enough times to pop all the pages in LinkedIn, or by closing and reopening the app. We might need to consult UX on how we should handle this: opening externally vs. opening inside the app (with perhaps a "Close" button to go back immediately).

I will add this to their to-do list.

**Review by m-rosario — CHANGES_REQUESTED** [review]

File: lib/widgets/screens/profile/profile_basic_info.dart [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+
+Widget _createVacationBanner(BuildContext context) {
+  final AppLocalizations l10n = AppLocalizations.of(context)!;
+  final ThemeData theme = Theme.of(context);
+  return Container(
+      color: theme.colorScheme.primaryContainer,
+      height: 48.0,
+      child: Row(mainAxisAlignment: MainAxisAlignment.center, children: [
+        Wrap(
+          crossAxisAlignment: WrapCrossAlignment.center,
```
[the remark this answers is not in the document]
Comment (ashlilili) [reply lib/widgets/screens/profile/profile_basic_info.dart]: ohh good catch!

File: lib/widgets/screens/profile/profile_basic_info.dart [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+
+  List<Widget> widgets = [
+    Text(
+      fullName,
+      style: theme.textTheme.titleMedium?.copyWith(
+        color: theme.colorScheme.onSurface,
+      ),
+    ),
+    if (pronouns != null)
+      Text(pronouns,
```
[the remark this answers is not in the document]
Comment (ashlilili) [reply lib/widgets/screens/profile/profile_basic_info.dart]: done!

File: lib/widgets/screens/profile/profile_basic_info.dart (hunk 2 above)
Comment (ashlilili) [reply lib/widgets/screens/profile/profile_basic_info.dart:86]: Great point! Added a comment below

File: lib/widgets/screens/profile/profile_basic_info.dart (hunk 3 above)
Comment (ashlilili) [reply lib/widgets/screens/profile/profile_basic_info.dart]: good catch, thanks! changed other icon colors too

File: lib/widgets/screens/profile/profile_basic_info.dart (hunk 4 above)
Comment (ashlilili) [reply lib/widgets/screens/profile/profile_basic_info.dart:11]: Makes sense!

**Comment by ashlilili** [comment]
Thanks @rupalg  and @m-rosario  for pointing out the behavior of launching url in mobile app.
I ran the code with android emulator and it's now opening the linkedin url in-app: 
inapp

----
alternatively, we could configure it to be opened in a browser as shown below
openinbrowser


---
I will follow up with Katy and Paris on this.

**Pull request merged**